### ISSUE TITLE
对于moe代码中的一个疑问

### ISSUE THREAD

**Issue by xiaomayi-ant** [jingyaogong/minimind#460]
<img width="921" height="425" alt="Image" src="https://github.com/user-attachments/assets/40efc7c5-cb7e-4249-aa9d-7dec0c2e6801" />

class MoEFeedForward：
           ....
          if self.training:
            x = x.repeat_interleave(self.config.num_experts_per_tok, dim=0)
            y = torch.empty_like(x, dtype=torch.float16)
            for i, expert in enumerate(self.experts):
                y[flat_topk_idx == i] = expert(x[flat_topk_idx == i]).to(y.dtype) 
            y = (y.view(*topk_weight.shape, -1) * topk_weight.unsqueeze(-1)).sum(dim=1) #这里是dim=1 还是dim=2
            y = y.view(*orig_shape)

thanks

**Comment by jingyaogong** [comment]
是dim=1
```
# Step 1: y.view(*topk_weight.shape, -1)
#   y: [2048, 512] -> [1024, 2, 512]
#   维度0: 1024个token
#   维度1: 每个token的2个专家
#   维度2: 512维特征

# Step 2: topk_weight.unsqueeze(-1)
#   [1024, 2] -> [1024, 2, 1]

# Step 3: 广播相乘（加权）
#   [1024, 2, 512] * [1024, 2, 1] = [1024, 2, 512]

# Step 4: sum(dim=1) - 沿专家维度求和
#   [1024, 2, 512] -> [1024, 512]
#   将每个token的2个专家输出加权聚合成1个输出
```

**Comment by xiaomayi-ant** [comment]
感谢您的回复，我理解了，thanks
